oak-auth-external/src/main/java/org/apache/jackrabbit/oak/spi/security/authentication/external/impl/ExternalLoginModule.java

```diff
@@ -228,7 +230,22 @@ public boolean login() throws LoginException {
             // before into the repository.
             UserManager userManager = getUserManager();
             SyncedIdentity sId = getSyncedIdentity(userId, userManager);
-
+            if (sId ==null && userManager != null && creds != null) {
+                // Check if the external user was registered with a different userId, and the same externalId
+                Object externalAttribute = credentialsSupport.getAttributes(creds).get(ExternalIdentityConstants.EXTERNAL_ID_ATTRIBUTE);
+                if (externalAttribute != null ) {
+                    @NotNull Iterator<Authorizable> authIterator = userManager.findAuthorizables(ExternalIdentityConstants.REP_EXTERNAL_ID, externalAttribute + ";" + idp.getName(), UserManager.SEARCH_TYPE_USER);
+                    if (authIterator.hasNext()) {
+                        //modify credentials to reflect the login name stored in oak
+                        Authorizable authorizable = authIterator.next();
+                        sId = getSyncedIdentity(authorizable.getID(), userManager);
+                        Map<String, ?> attributes = credentialsSupport.getAttributes(creds);
+                        HashMap<String, Object> newAttributes = new HashMap<>(attributes);
+                        newAttributes.put(AbstractLoginModule.SHARED_KEY_LOGIN_NAME, authorizable.getID());
```

Comment:
This class already inherits from `AbstractLoginModule` so it's not needed to add it while using the constant.

```suggestion
                        newAttributes.put(SHARED_KEY_LOGIN_NAME, authorizable.getID());
```